src/DI/ContainerBuilder.php

```diff
@@ -411,6 +417,16 @@ private function resolveImplement(ServiceDefinition $def, $name)
 	}
 
 
+	private function getTypeHintForParameter(\ReflectionParameter $param)
+	{
+		if (PHP_VERSION_ID >= 70000 && $param->hasType()) {
+			$type = $param->getType();
+			return $type->isBuiltin() ? (string)$type : '\\' . $type;
```

Comment:
`return $param->getType();` is IMHO enough. And probably functionality of getTypeHintForParameter() can be moved to `PhpReflection::getPropertyType()`.